### PR TITLE
fix: can not list the dashboards created in analyze

### PR DESCRIPTION
**Pull request by llmin**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

(describe what this merge request does)

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [ ] add new test cases
- [x] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [x] deploy with reporting

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

File: src/control-plane/backend/lambda/api/store/aws/quicksight.ts [hunk 1]
```diff
@@ -493,3 +509,154 @@ export const deleteDatasetOfPublishDashboard = async (
...
+  } while (nextToken);
+  return dashboardIds;
+};
+
+export const listDashboardsByApp = async (
+  region: string,
+  projectId: string,
+  appId: string,
+): Promise<IDashboard[]> => {
+  const folderId = `clickstream_${projectId}_${appId}`;
```
Comment (zxkane) [inline src/control-plane/backend/lambda/api/store/aws/quicksight.ts:627]: Let's share it with a method.

File: src/reporting/lambda/custom-resource/quicksight/index.ts [hunk 2]
```diff
@@ -12,6 +12,7 @@
  */
 
 import crypto from 'crypto';
+import { ResourceAlreadyExistsException } from '@aws-sdk/client-cloudwatch-events';
```
Comment (zxkane) [inline src/reporting/lambda/custom-resource/quicksight/index.ts:15]: Why using it from cloudwatch events?

File: src/reporting/lambda/custom-resource/quicksight/index.ts [hunk 3]
```diff
@@ -295,11 +302,37 @@ const createQuickSightDashboard = async (quickSight: QuickSight,
     },
   };
 
+  const folder = await quickSight.createFolder({
+    AwsAccountId: commonParams.awsAccountId,
+    FolderId: `clickstream_${commonParams.databaseName}_${commonParams.schema}`,
```
Comment (zxkane) [inline src/reporting/lambda/custom-resource/quicksight/index.ts:307]: Let's move it to a utility for cross-project sharing.

Comment (zxkane) [reply src/reporting/lambda/custom-resource/quicksight/index.ts:307]: We have similar usage in the backend's quicksight implementation. Do we have to share the code with the backend API? @tyyzqmf

File: src/reporting/private/iam.ts [hunk 4]
```diff
@@ -105,6 +105,23 @@ export function createRoleForQuicksightCustomResourceLambda(
       ],
     }),
 
+    new PolicyStatement({
+      effect: Effect.ALLOW,
+      resources: [
+        `${arnPrefix}:folder/clickstream_*`,
```
Comment (zxkane) [inline src/reporting/private/iam.ts:111]: Let's extract `clickstream_` as a constant named QS_FOLDER_PREFIX. And use it to generate the folder id.

Comment (tyyzqmf) [reply src/reporting/private/iam.ts]: Using existing shared variables: `QUICKSIGHT_RESOURCE_NAME_PREFIX`

File: src/reporting/lambda/custom-resource/quicksight/index.ts (hunk 3 above)
Comment (tyyzqmf) [reply src/reporting/lambda/custom-resource/quicksight/index.ts]: Use [QuickSight](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/Package/-aws-sdk-client-quicksight/Class/QuickSight/) . This way of calling services is more elegant by directly using 'QuickSightClient'.

File: src/common/constant.ts [hunk 5]
```diff
@@ -398,3 +399,60 @@ export const QUICKSIGHT_ANALYSIS_INFIX= '-analysis-';
 export const QUICKSIGHT_DATASET_INFIX= '-dataset-';
 
 export const SCAN_METADATA_WORKFLOW_PREFIX = 'ScanMetadataWorkflow';
+
+export const dataSetReaderPermissionActions = [
```
Comment (zxkane) [inline src/common/constant.ts:403]: consistent the variable name, like `DEFAULT_DASHBOARD_NAME_PREFIX`

**Review by zxkane — APPROVED** [review]

**Pull request merged**